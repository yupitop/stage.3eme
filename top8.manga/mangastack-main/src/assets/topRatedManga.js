export default [
  {
    title: 'One Piece',
    artist: ['Eiichiro Oda'],
    author: ['Eiichiro Oda'],
    description:
      'One Piece est un manga japonais créé par Eiichiro Oda',
    views: 735044,
    id: '700005',
    rating: {
      bayesian: '10.00',
    },
    mainCover: 'https://jumpg-assets.tokyo-cdn.com/secure/title/700005/title_thumbnail_portrait_list/326469.jpg?hash=ibIrwhI5h7XZ5__KnZMJnA&expires=2145884400',
  },
  {
    title: 'SAKAMOTO DAYS',
    artist: ['Yuto Suzuki'],
    author: ['Yuto Suzuki'],
    description:
      "Sakamoto Days est un manga japonais créé par ONE et illustré par Yuto Suzuki.",
    views: 9999999,
    id: '100127',
    rating: {
      bayesian: '10.00',
    },
    mainCover: 'https://jumpg-assets.tokyo-cdn.com/secure/title/100127/title_thumbnail_portrait_list/313621.jpg?hash=0HpK9-UWS7urxNciRiGxQQ&expires=2145884400',
  },
  {
    title: 'Bleach',
    artist: ['Tite Kubo'],
    author: ['Tite Kubo'],
    description:
      "Bleach est un manga japonais créé par Tite Kubo.",
    views: 1698463846313,
    id: '100004',
    rating: {
      bayesian: '10.00',
    },
    mainCover: 'https://jumpg-assets.tokyo-cdn.com/secure/title/100004/title_thumbnail_portrait_list/311788.jpg?hash=ypRnsOPrqQTWm-GBTUArRQ&expires=2145884400',
  },
  {
    title: 'Naruto',
    artist: ['Masashi Kishimoto'],
    author: ['Masashi Kishimoto'],
    description:
      "Naruto est un manga japonais créé par Masashi Kishimoto.",
    views: 10000000001,
    id: '100018',
    rating: {
      bayesian: '10.00',
    },
    mainCover: 'https://jumpg-assets.tokyo-cdn.com/secure/title/100018/title_thumbnail_portrait_list/313318.jpg?hash=vfQSOC-bbvsLAZyU8YSiQw&expires=2145884400',
  },
  
  
  {
    title:
      "JoJo's Bizarre Adventure Part 7 - Steel Ball Run (Official Colored)",
    artist: ['Araki Hirohiko'],
    author: ['Araki Hirohiko'],
    description:
      "In the American Old West, the world's greatest race is about to begin. Thousands line up in San Diego to travel over six thousand kilometers for a chance to win the grand prize of fifty million dollars. With the era of the horse reaching its end, contestants are allowed to use any kind of vehicle they wish. Competitors will have to endure grueling conditions, traveling up to a hundred kilometers a day through uncharted wastelands. The Steel Ball Run is truly a one-of-a-kind event.\r\n\r\nThe youthful Johnny Joestar, a crippled former horse racer, has come to San Diego to watch the start of the race. There he encounters Gyro Zeppeli, a racer with two steel balls at his waist instead of a gun. Johnny witnesses Gyro using one of his steel balls to unleash a fantastical power, compelling a man to fire his gun at himself during a duel. In the midst of the action, Johnny happens to touch the steel ball and feels a power surging through his legs, allowing him to stand up for the first time in two years. Vowing to find the secret of the steel balls, Johnny decides to compete in the race, and so begins his bizarre adventure across America on the Steel Ball Run.\r\n\r\nItalian summary:\r\n[Spoiler]Nel west americano, la gara pi&ugrave; grande e importante del mondo sta per iniziare. Migliaia di persone sono pronte a partire da San Diego per viaggiare pi&ugrave; di seimila chilometri incoraggiati da un premio enorme da cinquanta milioni di dollari. L'era dei cavalli sta per finire, i partecipanti allora possono usare il veicolo che preferiscono. I partecipanti devono superare situazioni estenuanti, viaggiando centinaia di chilometri al giorno passando per valli deserte inesplorate. La Steel Ball Run &egrave; unica nel suo genere.\r\n\r\nIl giovane Johnny Joestar, un ex corridore di cavalli disabile, &egrave; venuto a San Diego per vedere la partenza della gara. L&igrave; incontra Gyro Zeppelli, un corridore con due sfere d'acciaio pendenti fino alla sua vita, al posto della pistola. Johnny vede la sfera di Gyro e capisce della sua potenzialit&agrave;, riuscendo a far suicidare il nemico durante un duello. Nel mezzo dell'azione, Johnny tocca la sfera d'acciaio di Gyro e sente passare una potenza nelle sue gambe, questa potenza lo fa alzare in piedi per la prima volta dopo due anni. Dopo aver giurato di capire il segreto di quelle sfere, Johnny decide di partecipare alla gara, e qui iniziano le sue bizzarre avventure per tutta l'America nella Steel Ball Run.[/Spoiler]",
    views: 1622113,
    id: 6272,
    rating: {
      bayesian: '9.33',
    },
    mainCover: 'https://fr.web.img3.acsta.net/c_310_420/pictures/20/03/10/12/07/4001550.jpg',
  },
  
  {
    title: 'Berserk',
    artist: ['Miura Kentaro'],
    author: ['Miura Kentaro'],
    description:
      'Guts, known as the Black Swordsman, seeks sanctuary from the demonic forces attracted to him and his woman because of a demonic mark on their necks, and also vengeance against the man who branded him as an unholy sacrifice. Aided only by his titanic strength gained from a harsh childhood lived with mercenaries, a gigantic sword, and an iron prosthetic left hand, Guts must struggle against his bleak destiny, all the while fighting with a rage that might strip him of his humanity.\r\n\r\n[u][*]Won the Award for Excellence at the sixth installment of Tezuka Osamu Cultural Prize in 2002.[/u]\r\n[b]Links to AnimeNewsNetwork:[/b]\r\nMovies:\r\n[*][url=https://www.animenewsnetwork.com/encyclopedia/anime.php?id=12397]The Golden Age Arc I[/url]\r\n[*][url=https://www.animenewsnetwork.com/encyclopedia/anime.php?id=13799]The Golden Age Arc II[/url]\r\n[*][url=https://www.animenewsnetwork.com/encyclopedia/anime.php?id=13800]The Golden Age Arc III[/url]\r\nAnime (2015):\r\n[*][url=https://www.animenewsnetwork.com/encyclopedia/anime.php?id=328]Season 1[/url]\r\n[*][url=https://www.animenewsnetwork.com/encyclopedia/anime.php?id=18160]Season 2[/url]\r\n[*][url=https://www.animenewsnetwork.com/encyclopedia/anime.php?id=19015]Season 3[/url]\r\n\r\n[url=https://shopee.co.th/berserk%E0%B9%80%E0%B8%9A%E0%B8%AD%E0%B8%A3%E0%B9%8C%E0%B9%80%E0%B8%8B%E0%B8%B4%E0%B8%A3%E0%B9%8C%E0%B8%81-%E0%B8%9A%E0%B8%B4%E0%B9%8A%E0%B8%81%E0%B8%9A%E0%B8%B8%E0%B8%84-%E0%B9%81%E0%B8%A2%E0%B8%81%E0%B9%80%E0%B8%A5%E0%B9%88%E0%B8%A1-1-20-%E0%B9%83%E0%B8%AB%E0%B8%A1%E0%B9%88%E0%B8%A1%E0%B8%B7%E0%B8%AD%E0%B8%AB%E0%B8%99%E0%B8%B6%E0%B9%88%E0%B8%87-i.63434898.1855330863] Official Thai [/url]',
    views: 2433186,
    id: '801513ba-a712-498c-8f57-cae55b38cc92',
    rating: {
      bayesian: '9.37',
    },
    mainCover: 'https://cdn.myanimelist.net/images/manga/1/157931.jpg',
  },
  
  {
    title: 'The Horizon',
    artist: ['Jeong Ji-Hoon'],
    author: ['Jeong Ji-Hoon'],
    description:
      "Two children unexpectedly meet in the midst of war. After running away from the chaos, they come across a long empty road. With no adults to rely on, the two strangers, now become friends, walk alongside each other to see what's at the end of the road. All they hope is to keep being able to move forward. Just what kind of trials and tragedy awaits them during their journey for survival?\r\n[hr][b]Links[/b]:\r\n[*][url=https://www.webtoons.com/th/drama/horizon/list?title_no=2169]Official Thai Translation[/url]\r\n[*][url=https://www.kuaikanmanhua.com/web/topic/1853/]Official Simplified Chinese Translation[/url]\r\n\r\n[*][url=https://www.aladin.co.kr/shop/wproduct.aspx?ItemId=97911543]Original Book (Aladin)[/url]\r\n[*][url=http://www.yes24.com/Product/Goods/33667823]Original Book (Yes24)[/url]\r\n[*][url=https://series.naver.com/comic/detail.nhn?productNo=4458915]Original Book (Naver)[/url]",
    views: 115181,
    id: 47562,
    rating: {
      bayesian: '9.35',
    },
    mainCover: 'https://mangadex.org/images/manga/47562.jpg',
  },
  {
    title: 'Yotsuba to!',
    artist: ['Azuma Kiyohiko'],
    author: ['Azuma Kiyohiko'],
    description:
      "Yotsuba is a strange little girl with a big personality! Even in the most trivial, unremarkable encounters, Yotsuba's curiosity and enthusiasm quickly turns the everyday into the extraordinary!\r\n2016 Osamu Tezuka Culture Award - grand prize. Yotsuba&amp;! received an Excellence Award for Manga at the 2006 Japan Media Arts Festival. In 2008 Yotsuba&amp;! was nominated for the 12th Osamu Tezuka Culture Award and the Eisner Award &quot;Best Publication for Kids&quot; category, but did not win either, and was runner-up for the first annual Manga Taisho award.",
    views: 785434,
    id: 311,
    rating: {
      bayesian: '9.35',
    },
    mainCover: 'https://mangadex.org/images/manga/311.jpg',
  },
  {
    title: 'Kusuriya no Hitorigoto',
    artist: ['Nekokurage'],
    author: ['Hyuuga Natsu', 'Nanao Ikki'],
    description:
      "In the imperial court, a young woman is put into servitude, Maomao. The tale is just beginning for the woman pharmacist from the red-light district, as rumors circulate about the emperor's children's lives being short-lived. Her curious nature and thirst for knowledge pushes her to action. To satisfy her curiosity, this young pharmacist will investigate the origin! What is shifting in the imperial court&hellip;?!\r\n\r\n[*][u]Won the Grand Prize in the Printed Manga category at the Tsugi ni Kuru Manga Award 2019.[/u]\r\n[hr][u][b]Italian / Italiano:[/b][/u]\r\n[spoiler] Giovani donne vengono spesso vendute per lavorare per un breve periodo nell'harem dell'imperatore. &Egrave; questo il caso di Maomao, farmacista presso il quartiere a luci rosse. Non ha alcun interesse nel farsi notare a palazzo e preferisce starsene per conto suo a preparare medicine, studiare veleni o fare esperimenti su se stessa. Ma quando a corte si diffonde la voce che i figli dell'imperatore sono destinati a una morte prematura, Maomao non riesce a rimanere indifferente. Mentre cerca di indagare, cattura l'attenzione di misterioso e affascinante ufficiale di alto rango. Ha cos&igrave; inizio la sua particolare carriera.[/spoiler]\r\n[hr][u][b]Russian / Русский:[/b][/u]\r\n[spoiler]В императорский двор была доставлена молодая девушка Маомао. История её берёт начало в районе красных фонарей, где начали ходить слухи о том, что жизни детей императора под угрозой. Ею движут любопытство и жажда знаний. Чтобы удовлетворить своё любопытство, молодая целительница будет докапываться до истины! Что же изменится при императорском дворе?! Адаптация одноимённого ранобэ.[/spoiler]\r\n[hr][b][u]Links:[/b]\r\n[*][url=https://twitter.com/nekokurage_]Mangaka's Twitter[/url]\r\n[*][url=https://bookwalker.jp/series/160955/]Japanese Light Novel Bookwalker[/url]\r\n[*][url=https://j-novel.club/s/the-apothecary-diaries/]Official English Light Novel Translation[/url]\r\n[*][url=https://manga.bilibili.com/detail/mc28218]Official Chinese (Simplified) Translation[/url]",
    views: 2408773,
    id: 21562,
    rating: {
      bayesian: '9.33',
    },
    mainCover: 'https://mangadex.org/images/manga/21562.jpg',
  },
  {
    title: 'Akatsuki no Yona',
    artist: ['Kusanagi Mizuho'],
    author: ['Kusanagi Mizuho'],
    description:
      "Yona is the sole princess of the Kingdom of Kouka, living a luxurious and carefree life as a princess should. She has it all: the finest clothes and cosmetics, the most divine sweets, a loving emperor as a father, and the hottest cousin crush, Soo-won, any girl could ever hope to have. Now, if only her bodyguard, Son Hak, weren't so annoying towards her, and if her hair weren't so red...\r\nBut her nearly perfect world quickly shatters as the man she loves, Soo-won, murders her father and the path to his ascension to the throne is assured. Son Hak escapes with Yona, and they lead a life on the run from that point on.\r\n\r\n[hr][b][u]Russian / Русский:[/u][/b]\r\n[spoiler]Йона &mdash; единственная принцесса королевства Коука. Она ведет роскошную и беззаботную жизнь, как и подобает принцессам. У неё есть всё: лучшая одежда и косметика, превосходные сладости, любящий отец Император Иль, и сильнейшие чувства к кузену Су Вону, о котором многие могли лишь мечтать. Вот только если бы её телохранитель Сон Хак не так раздражал и её волосы не были такими красными. \r\nВсе меняется в ночь на шестнадцатый день рождения девушки. Почти идеальный мир быстро рассыпается, когда человек, которого она любит &mdash; Су Вон &mdash; убил её отца. Путь к его становлению на престол ныне гарантирован. Вместе с Хаком, Йона скрылась из замка в поисках помощи, и с этого момента они ведут жизнь в бегах&hellip;[/spoiler]",
    views: 1250081,
    id: 3996,
    rating: {
      bayesian: '9.33',
    },
    mainCover: 'https://mangadex.org/images/manga/3996.jpg',
  },
  {
    title: 'Houseki no Kuni',
    artist: ['Ichikawa Haruko'],
    author: ['Ichikawa Haruko'],
    description:
      "In the distant future, a new life form called Houseki (gems) are born. The 28 Houseki must fight against the moon dwellers who want to attack them and turn them into decorations, thus each gem is assigned a role such as a fighter or a medic. Though they hope to fight the moon dwellers, Phos is a gem who is given no assignment until the gems' master Adamantine asks them to write an encyclopedia.",
    views: 1299954,
    id: 10749,
    rating: {
      bayesian: '9.33',
    },
    mainCover: 'https://mangadex.org/images/manga/10749.jpeg',
  },
  
];
