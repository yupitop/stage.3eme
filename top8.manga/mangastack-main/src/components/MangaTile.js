import { Link } from 'react-router-dom';
import {
  Typography, Card, CardContent,
  CardMedia, CardActionArea
} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { Rating } from '@material-ui/lab';
import { htmlDecode } from '../utils/utils';

export const getShortText = (description, cutoff) => {
  const split = htmlDecode(description).replace(/\[.*?\]/g, '');
  if (split.length <= cutoff) {
    return split;
  } else {
    return split.slice(0, cutoff) + '...';
  }
};

export const getShortAuthors = (author, artist) => {
  const cutoff = 30
  const authors = htmlDecode(Array.from(new Set(author.concat(artist))).join(', '));
  if (authors.length <= cutoff) {
    return authors;
  } else {
    return authors.slice(0, cutoff) + '...';
  }
};

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    height: 300,
    width: '100%'
  },
  details: {
    display: 'flex',
    flexDirection: 'column',
  },
  content: {
    flex: '1 0 auto',
  },
  cover: {
    minWidth: 150,
    [theme.breakpoints.only('xs')]: {
      minWidth: 50
    },
    [theme.breakpoints.only('sm')]: {
      minWidth: 100
    }
  }
}));

const MangaTile = ({ id, title, artist, author, description, rating, views, mainCover }) => {
  const classes = useStyles();
  return (
    <Card className={classes.root} elevation={10}>
      <CardActionArea>
        <Link to={`/manga/${id}`}>
          <CardMedia
            component="img"
            height={300}
            className={classes.cover}
            image={mainCover}
            title={`Cover for ${title}`}
          />
        </Link>
      </CardActionArea>
      <div className={classes.details}>
        <CardContent className={classes.content}>
          <Link
            to={`/manga/${id}`}
            style={{ color: 'inherit', textDecoration: 'none' }}
          >
            <Typography component="h6" variant="h6">
              {getShortText(title, 50)}
            </Typography>
          </Link>
          <Rating value={rating.bayesian / 2} precision={0.5} readOnly />
          <Typography>
            {views.toLocaleString()} views
              </Typography>
          <Typography variant="subtitle1" color="textSecondary">
            {getShortAuthors(author, artist)}
          </Typography>
          <Typography variant="body2">
            {getShortText(description, 150) || 'No synopsis available.'}
          </Typography>
        </CardContent>
      </div>
    </Card>
  );
};

export default MangaTile;