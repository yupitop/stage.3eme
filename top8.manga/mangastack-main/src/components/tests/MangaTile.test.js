import { MemoryRouter, Route } from 'react-router-dom';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import MangaTile, { getShortText, getShortAuthors } from '../MangaTile';
import topRatedManga from '../../assets/topRatedManga';

let container;
let mockManga;
let component;
let testLocation;

beforeEach(() => {
  mockManga = topRatedManga[0];
  component = (
    <MemoryRouter initialEntries={['/']}>
      <MangaTile {...mockManga} />
      <Route 
        path="*"
        render={({ location }) => {
          testLocation = location;
          return null;
        }}
      />
    </MemoryRouter>
  );
  container = render(component).container;
});

afterEach(cleanup);

it('should render', () => {
  expect(container).toMatchSnapshot();
});

it('displays title, synopsis, authors, and views', () => {
  const title = screen.getByText(getShortText(mockManga.title, 50));
  expect(title).toBeInTheDocument();

  const synopsis = screen.getByText(getShortText(mockManga.description, 150));
  expect(synopsis).toBeInTheDocument();
  
  const authors = screen.getByText(getShortAuthors(mockManga.author, mockManga.artist));
  expect(authors).toBeInTheDocument();
  
  const views = screen.getByText(`${mockManga.views.toLocaleString()} views`);
  expect(views).toBeInTheDocument();
});

it('displays cover image', () => {
  const image = container.querySelector(`img[title="Cover for ${mockManga.title}"]`);
  expect(image.getAttribute('src')).toBe(mockManga.mainCover);
});

it('title links to the manga page', () => {
  const title = screen.getByText(getShortText(mockManga.title, 50));
  fireEvent.click(title);
  expect(testLocation.pathname).toBe(`/manga/${mockManga.id}`);
});

it('main cover links to the manga page', () => {
  const image = container.querySelector(`img[title="Cover for ${mockManga.title}"]`);
  fireEvent.click(image);
  expect(testLocation.pathname).toBe(`/manga/${mockManga.id}`);
});
