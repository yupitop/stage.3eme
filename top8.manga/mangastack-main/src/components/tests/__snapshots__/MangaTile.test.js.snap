// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation10 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <a
        href="/manga/48045"
      >
        <img
          class="MuiCardMedia-root makeStyles-cover-4 MuiCardMedia-media MuiCardMedia-img"
          height="300"
          src="https://mangadex.org/images/manga/48045.jpg"
          title="Cover for Sousou no Frieren"
        />
      </a>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="makeStyles-details-2"
    >
      <div
        class="MuiCardContent-root makeStyles-content-3"
      >
        <a
          href="/manga/48045"
          style="text-decoration: none;"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Sousou no Frieren
          </h6>
        </a>
        <span
          aria-label="4.5 Stars"
          class="MuiRating-root MuiRating-readOnly"
          role="img"
        >
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal"
          >
            <span
              style="width: 50%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          2,993,789
           views
        </p>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Yamada Kanehito, Abe Tsukasa
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          The adventure is over but life goes on for an elf mage just beginning to learn what living is all about. Elf mage Frieren and her courageous fellow ad...
        </p>
      </div>
    </div>
  </div>
</div>
`;
